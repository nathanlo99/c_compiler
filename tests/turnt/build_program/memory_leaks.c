
int wain(int a, int b) {
  // Create 'a' memory leaks, each pf size 'b'
  int i = 0;
  int *ptr = NULL;
  while (i < a) {
    ptr = new int[b];
    i = i + 1;
  }
  return 0;
}
