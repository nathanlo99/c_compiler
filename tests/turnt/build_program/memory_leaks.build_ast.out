Procedure {
  name: wain
  return_type: int
  parameters: 
    a: int (used)
    b: int (used)
  declarations: 
    i: int = 0 (used)
    ptr: int* = 1 (used)
  statements: 
    WhileStatement {
      condition: 
        BinaryExpr {
          lhs: 
            VariableExpr(i) : int
          operation: LessThan
          rhs: 
            VariableExpr(a) : int
        } : int
      body: 
        ExprStatement {
          AssignmentExpr {
            lhs: 
              VariableLValueExpr(ptr) : int*
            rhs: 
              NewExpr {
                VariableExpr(b) : int
              } : int*
          } : int*
        }
        ExprStatement {
          AssignmentExpr {
            lhs: 
              VariableLValueExpr(i) : int
            rhs: 
              BinaryExpr {
                lhs: 
                  VariableExpr(i) : int
                operation: Add
                rhs: 
                  1: int
              } : int
          } : int
        }
    }
  return_expr: 
    0: int
}

